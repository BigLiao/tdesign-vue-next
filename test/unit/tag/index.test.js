import { mount } from '@vue/test-utils';
import Tag from '@/src/tag/index.ts';
import CheckTag from '@/src/tag/check-tag.tsx';

describe('Tag or CheckTag', () => {
  describe(':Tag:props', () => {
    it(':theme:', () => {
      const wrapper = mount({
        render() {
          return <Tag></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':theme:default', () => {
      const wrapper = mount({
        render() {
          return <Tag theme={'default'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':theme:primary', () => {
      const wrapper = mount({
        render() {
          return <Tag theme={'primary'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':theme:info', () => {
      const wrapper = mount({
        render() {
          return <Tag theme={'info'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':theme:warning', () => {
      const wrapper = mount({
        render() {
          return <Tag theme={'warning'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':theme:danger', () => {
      const wrapper = mount({
        render() {
          return <Tag theme={'danger'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':theme:success', () => {
      const wrapper = mount({
        render() {
          return <Tag theme={'success'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':size:small', () => {
      const wrapper = mount({
        render() {
          <Tag size='small'></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':size:medium', () => {
      const wrapper = mount({
        render() {
          <Tag size="medium"></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':size:large', () => {
      const wrapper = mount({
        render() {
          return <Tag size={'large'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':icon:iconfont-string:home', () => {
      const wrapper = mount({
        render() {
          return <Tag icon="home"></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':closable', () => {
      const fn = jest.fn();
      const wrapper = mount({
        render() {
          return <Tag closable onClose={fn}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
      wrapper.find('.t-icon-close').trigger('click');
      expect(fn).toHaveBeenCalled();
    });

    it(':checked', () => {
      const wrapper = mount({
        render() {
          return <Tag checked></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':disabled', () => {
      const fn = jest.fn();
      const wrapper = mount({
        render() {
          return <Tag disabled onClick={fn}></Tag>;
        },
      });
      expect(fn).not.toHaveBeenCalled();
      expect(wrapper).toMatchSnapshot();
    });

    it(':effect:dark;theme:default', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="dark" theme={'default'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:dark;theme:primary', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="dark" theme={'primary'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:dark;theme:info', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="dark" theme={'info'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:dark;theme:warning', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="dark" theme={'warning'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:dark;theme:danger', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="dark" theme={'danger'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:dark;theme:success', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="dark" theme={'success'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':effect:plain;theme:default', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="plain" theme={'default'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:plain;theme:primary', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="plain" theme={'primary'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:plain;theme:info', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="plain" theme={'info'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:plain;theme:warning', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="plain" theme={'warning'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:plain;theme:danger', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="plain" theme={'danger'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:plain;theme:success', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="plain" theme={'success'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:light;theme:default', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="light" theme={'default'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:light;theme:primary', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="light" theme={'primary'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:light;theme:info', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="light" theme={'info'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:light;theme:warning', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="light" theme={'warning'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:light;theme:danger', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="light" theme={'danger'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':effect:light;theme:success', () => {
      const wrapper = mount({
        render() {
          return <Tag effect="light" theme={'success'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':shape:round', () => {
      const wrapper = mount({
        render() {
          return <Tag shape={'round'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':shape:square', () => {
      const wrapper = mount({
        render() {
          return <Tag shape={'square'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':shape:mark', () => {
      const wrapper = mount({
        render() {
          return <Tag shape={'mark'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':maxWidth', () => {
      const wrapper = mount({
        render() {
          return <Tag max-width={100}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':icon', () => {
      const wrapper = mount({
        render() {
          return <Tag icon={'search'}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('CheckTag:props', () => {
    it(':checked', () => {
      const wrapper = mount({
        render() {
          return <CheckTag checked></CheckTag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':disabled', () => {
      const fn = jest.fn();
      const wrapper = mount({
        render() {
          return <CheckTag disabled onClick={fn}></CheckTag>;
        },
      });
      expect(fn).not.toHaveBeenCalled();
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Tag:slot', () => {
    it('<icon>', () => {
      const wrapper = mount(Tag, {
        scopedSlots: {
          icon: '<div></div>',
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('@event: Tag', () => {
    it('Event passthrough: click', () => {
      const fn = jest.fn();
      const wrapper = mount({
        render() {
          return <Tag onClick={fn}>text</Tag>;
        },
      });
      wrapper.find(Tag).trigger('click');
      expect(fn).toHaveBeenCalled();
    });
    it('Event passthrough: close', () => {
      const fn = jest.fn();
      const wrapper = mount({
        render() {
          return <Tag closable onClose={fn}></Tag>;
        },
      });
      expect(wrapper).toMatchSnapshot();
      wrapper.find('.t-icon-close').trigger('click');
      expect(fn).toHaveBeenCalled();
    });
  });

  describe('@event: CheckTag', () => {
    it('Event passthrough: change', () => {
      const fn = jest.fn();
      const wrapper = mount({
        render() {
          return <CheckTag checked onChange={fn}>text</CheckTag>;
        },
      });
      wrapper.find(CheckTag).trigger('click');
      expect(fn).toBeCalled();
    });
  });
});
