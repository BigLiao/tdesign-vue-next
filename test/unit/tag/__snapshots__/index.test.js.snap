// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag or CheckTag :Tag:props :checked 1`] = `<span class="t-tag t-tag--default t-tag--dark t-tag--checked"></span>`;

exports[`Tag or CheckTag :Tag:props :closable 1`] = `<span class="t-tag t-tag--default t-tag--dark t-tag--close"><i class="t-icon t-icon-close"></i></span>`;

exports[`Tag or CheckTag :Tag:props :disabled 1`] = `<span class="t-tag t-tag--default t-tag--dark t-tag--disabled"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:dark;theme:danger 1`] = `<span class="t-tag t-tag--danger t-tag--dark"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:dark;theme:default 1`] = `<span class="t-tag t-tag--default t-tag--dark"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:dark;theme:info 1`] = `<span class="t-tag t-tag--info t-tag--dark"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:dark;theme:primary 1`] = `<span class="t-tag t-tag--primary t-tag--dark"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:dark;theme:success 1`] = `<span class="t-tag t-tag--success t-tag--dark"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:dark;theme:warning 1`] = `<span class="t-tag t-tag--warning t-tag--dark"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:light;theme:danger 1`] = `<span class="t-tag t-tag--danger t-tag--light"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:light;theme:default 1`] = `<span class="t-tag t-tag--default t-tag--light"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:light;theme:info 1`] = `<span class="t-tag t-tag--info t-tag--light"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:light;theme:primary 1`] = `<span class="t-tag t-tag--primary t-tag--light"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:light;theme:success 1`] = `<span class="t-tag t-tag--success t-tag--light"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:light;theme:warning 1`] = `<span class="t-tag t-tag--warning t-tag--light"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:plain;theme:danger 1`] = `<span class="t-tag t-tag--danger t-tag--plain"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:plain;theme:default 1`] = `<span class="t-tag t-tag--default t-tag--plain"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:plain;theme:info 1`] = `<span class="t-tag t-tag--info t-tag--plain"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:plain;theme:primary 1`] = `<span class="t-tag t-tag--primary t-tag--plain"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:plain;theme:success 1`] = `<span class="t-tag t-tag--success t-tag--plain"></span>`;

exports[`Tag or CheckTag :Tag:props :effect:plain;theme:warning 1`] = `<span class="t-tag t-tag--warning t-tag--plain"></span>`;

exports[`Tag or CheckTag :Tag:props :icon 1`] = `<span class="t-tag t-tag--default t-tag--dark"><i class="t-icon t-icon-search"></i></span>`;

exports[`Tag or CheckTag :Tag:props :icon:iconfont-string:home 1`] = `<span class="t-tag t-tag--default t-tag--dark"><i class="t-icon t-icon-home"></i></span>`;

exports[`Tag or CheckTag :Tag:props :maxWidth 1`] = `<span class="t-tag t-tag--default t-tag--dark t-tag--ellipsis" style="max-width: 100px;"></span>`;

exports[`Tag or CheckTag :Tag:props :shape:mark 1`] = `<span class="t-tag t-tag--default t-tag--dark t-tag--mark"></span>`;

exports[`Tag or CheckTag :Tag:props :shape:round 1`] = `<span class="t-tag t-tag--default t-tag--dark t-tag--round"></span>`;

exports[`Tag or CheckTag :Tag:props :shape:square 1`] = `<span class="t-tag t-tag--default t-tag--dark"></span>`;

exports[`Tag or CheckTag :Tag:props :size:large 1`] = `<span class="t-tag t-tag--default t-size-l t-tag--dark"></span>`;

exports[`Tag or CheckTag :Tag:props :size:medium 1`] = ``;

exports[`Tag or CheckTag :Tag:props :size:small 1`] = ``;

exports[`Tag or CheckTag :Tag:props :theme: 1`] = `<span class="t-tag t-tag--default t-tag--dark"></span>`;

exports[`Tag or CheckTag :Tag:props :theme:danger 1`] = `<span class="t-tag t-tag--danger t-tag--dark"></span>`;

exports[`Tag or CheckTag :Tag:props :theme:default 1`] = `<span class="t-tag t-tag--default t-tag--dark"></span>`;

exports[`Tag or CheckTag :Tag:props :theme:info 1`] = `<span class="t-tag t-tag--info t-tag--dark"></span>`;

exports[`Tag or CheckTag :Tag:props :theme:primary 1`] = `<span class="t-tag t-tag--primary t-tag--dark"></span>`;

exports[`Tag or CheckTag :Tag:props :theme:success 1`] = `<span class="t-tag t-tag--success t-tag--dark"></span>`;

exports[`Tag or CheckTag :Tag:props :theme:warning 1`] = `<span class="t-tag t-tag--warning t-tag--dark"></span>`;

exports[`Tag or CheckTag @event: Tag Event passthrough: close 1`] = `<span class="t-tag t-tag--default t-tag--dark t-tag--close"><i class="t-icon t-icon-close"></i></span>`;

exports[`Tag or CheckTag CheckTag:props :checked 1`] = `<span class="t-tag t-tag--check t-tag--default t-tag--checked"></span>`;

exports[`Tag or CheckTag CheckTag:props :disabled 1`] = `<span class="t-tag t-tag--check t-tag--default t-tag--disabled"></span>`;

exports[`Tag or CheckTag Tag:slot <icon> 1`] = `<span class="t-tag t-tag--default t-tag--dark"></span>`;
